src/burn/engine/registration.cpp

```diff
@@ -642,9 +644,15 @@ extern "C" HRESULT RegistrationSessionBegin(
         hr = RegWriteStringArray(hkRegistration, BURN_REGISTRATION_REGISTRY_BUNDLE_PATCH_CODE, pRegistration->rgsczPatchCodes, pRegistration->cPatchCodes);
         ExitOnFailure(hr, "Failed to write %ls value.", BURN_REGISTRATION_REGISTRY_BUNDLE_PATCH_CODE);
 
-        hr = RegWriteStringFormatted(hkRegistration, BURN_REGISTRATION_REGISTRY_BUNDLE_VERSION, L"%hu.%hu.%hu.%hu", (WORD)(pRegistration->qwVersion >> 48), (WORD)(pRegistration->qwVersion >> 32), (WORD)(pRegistration->qwVersion >> 16), (WORD)(pRegistration->qwVersion));
+        hr = RegWriteStringFormatted(hkRegistration, BURN_REGISTRATION_REGISTRY_BUNDLE_VERSION, L"%hu.%hu.%hu.%hu", (WORD) (pRegistration->qwVersion >> 48), (WORD) (pRegistration->qwVersion >> 32), (WORD) (pRegistration->qwVersion >> 16), (WORD) (pRegistration->qwVersion));
```

Comment:
Generally, `static_cast` and friends are preferred. I gave up fighting Fredrik in some parts of Burn way back when.